### PR TITLE
refactoring code; add special formatter

### PR DESCRIPTION
**Pull request by ivanuzk**
- code polishing
- refactoring to make code compact and concise
- adds special formatting rules

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use_try_shorthand = true
+binop_separator = "Front"
+use_small_heuristics = "Max"
+fn_args_layout = "Compressed"
+
+# Nightly
+indent_style = "Visual"
+imports_indent = "Visual"
+merge_imports = true
+wrap_comments = true
```
Comment (sigrlami) [inline rustfmt.toml:17]: I think that's strange, it messes up Emacs setup, can you hide temporary?

File: src/client/did.rs [hunk 2]
```diff
@@ -64,8 +64,8 @@ for service in self.services {
 self._check_alias_is_unique_and_add_to_used(self.used_service_aliases, service.alias);
 }
 }
-fn __repr__<RT>(&self) -> RT {
-return "<{}.{} (management_keys={}, did_keys={}, services={})>".format(self.__module__, type_(self).__name__, self.management_keys.len(), self.did_keys.len(), self.services.len());
+fn represent<RT>(&self) -> RT {
+return "<{}.{} (management_keys={}, did_keys={}, services={})>".format(self.module, type_(self).__name__, self.management_keys.len(), self.did_keys.len(), self.services.len());
```
Comment (sigrlami) [inline src/client/did.rs:68]: I would prefer ditching self at all and refer to internal representation, look if that possible

**Review by sigrlami — CHANGES_REQUESTED** [review]
Overall good, need small tweaks

**Comment by ivanuzk** [comment]
merging based on external resolution

**Pull request merged**